docs/topics/lombok.md

```diff
@@ -0,0 +1,149 @@
+
+> Kotlin compiler ignores Lombok annotations if you use them in Kotlin code.
+>
+{type="note"}
+
+Please read below about how to apply the Lombok Kotlin compiler plugin to your Gradle/Maven build.
+
+## Gradle
+
+### Main settings
+
+Apply the `kotlin-plugin-lombok` Gradle plugin:
+
+```groovy
+plugins {
+    id "org.jetbrains.kotlin.plugin.lombok" version "%kotlinVersion%"
+}
+```
+
+Alternatively, you can use the `apply plugin` syntax:
+
+```groovy
+apply plugin: 'kotlin-plugin-lombok'
+```
+
+The Lombok itself should be set up as the Java annotation processor (not via kapt) or as the plugin as below:
+
+```groovy
+plugins {
+    id "io.freefair.lombok" version "5.3.0"
+}
+```
+Learn the [test project with examples of Lombok compiler plugin usage](https://github.com/kotlin-hands-on/kotlin-lombok-examples/tree/master/kotlin_lombok_gradle/nokapt).
+
+### Defining the place of lombok.config
+
+If you use the file `lombok.config` for [configuring Lombok](https://projectlombok.org/features/configuration),
+Kotlin Lombok compiler plugin should know where this file situates. Use the following syntax in your `build.gradle` file:
+
+```groovy
+kotlinLombok {
+    lombokConfigurationFile(file("lombok.config"))
+}
+```
+
+Learn the [test project with examples of Lombok compiler plugin and `lombok.config` usages](https://github.com/kotlin-hands-on/kotlin-lombok-examples/tree/master/kotlin_lombok_gradle/withconfig).
+
+## Maven
+
+Here's how to use Lombok Kotlin compiler plugin with Maven:
+
+```xml
+<plugin>
+    <groupId>org.jetbrains.kotlin</groupId>
+    <artifactId>kotlin-maven-plugin</artifactId>
+    <version>${kotlin.version}</version>
+    <configuration>
+        <compilerPlugins>
+            <plugin>lombok</plugin>
+        </compilerPlugins>
+        <pluginOptions>
+            <option>lombok:config=${project.basedir}/lombok.config</option>
+        </pluginOptions>
+    </configuration>
+    <dependencies>
+        <dependency>
+            <groupId>org.jetbrains.kotlin</groupId>
+            <artifactId>kotlin-maven-lombok</artifactId>
+            <version>${kotlin.version}</version>
+        </dependency>
+    </dependencies>
+</plugin>
+```
+
+Please refer to the [Gradle](#gradle) section for detailed information about the Lombok Kotlin compiler plugin configuration.
+
+Learn the [test project example of Lombok compiler plugin and `lombok.config` usages](https://github.com/kotlin-hands-on/kotlin-lombok-examples/tree/master/kotlin_lombok_maven/nokapt).
+
+## Using the plugin along with kapt
+
+If you use [kapt](kapt.md) compiler plugin, note that Java's Lombok should be launched via annotation processor of Java's compiler.
+By default, kapt assumes that all annotation processors work through it and disables Java's compiler's annotation processors.
+To make Lombok working, you need to change this behavior.
+
+If you use the Gradle build system, add the additional flag to the `build.gradle` file to make Java's compiler's annotations
```

Comment:
"the Gradle build system" why not "Gradle"?
"add the additional" - repetition
"flag" - config property or option

Reply:
>"add the additional" - repetition

Ha-ha, even didn't catch it from the first time – was thinking where is the repetition and what's wrong)) Personally, I don't see anything bad in _add the additional_. They are just words with similar letters, no?

Actually, we have such wording in our kapt.md:
>Add the additional flag to the `build.gradle` file to enable error type inferring in stubs:

Will fix it too.

I need to say that my experience tells me that developers usually call things with true/false values flags.

Reply:
> They are just words with similar letters, no?

No, these are cognate words - they share the same root. **add**itional is what was **add**ed.